recipe/build.sh

```diff
@@ -3,15 +3,43 @@
 mkdir build
 cd build
 
+# OS X
+# ----
+#
+# On MacOSX, the SDK is not distributed by conda
+#
+# Until version 10.13 (included), the OSX SDK headers were in
+# /usr/include/ and /usr/include/CXX_VERSION for the standard
+# library headers.
+#
+# From version 10.14, the OSX SDK is installed in
+# /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/
+#
+# Linux
+# -----
+#
+# The host compiler is distributed as a conda package.
+#
+
+EXTRA_CMAKE_ARGS=""
+if [[ `uname` == "Darwin" ]]; then
+  EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS} -DCLING_INCLUDE_PATHS=/usr/include:/usr/include/c++/$(${CXX} -dumpversion):/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include:/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1"
+else
+  EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS} -DCLING_INCLUDE_PATHS=${PREFIX}/${HOST}/include/c++/$(${CXX} -dumpversion):${PREFIX}/${HOST}/sysroot/usr/include"
```

Comment:
You might want to use `$CXX -print-sysroot` to discover the sysroot and evaluate paths relative to that.

If you want *all* the include paths, `$CXX -E -x c++ - -v < /dev/null` will show them, if not in the easiest to parse format:

```
Reading specs from /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=/opt/conda/envs/gcc/bin/x86_64-conda_cos6-linux-gnu-c++
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/gcc_built --with-sysroot=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.449-a04d0' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.449-a04d0)
COLLECT_GCC_OPTIONS='-E' '-v' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
 /opt/conda/envs/gcc/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/cc1plus -E -quiet -v -iprefix /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/ -isysroot /opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot -D_GNU_SOURCE - -mtune=generic -march=x86-64
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0"
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu"
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward"
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include"
ignoring nonexistent directory "/opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/home/rdonnelly/mc/conda-bld/compilers_linux-64_1534865402226/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot/include"
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include-fixed"
ignoring duplicate directory "/opt/conda/envs/gcc/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here: *** < start here
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include-fixed
 /opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include
 /opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/include
End of search list.  **** < end here
COMPILER_PATH=/opt/conda/envs/gcc/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/:/opt/conda/envs/gcc/bin/../libexec/gcc/:/opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/
LIBRARY_PATH=/opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/:/opt/conda/envs/gcc/bin/../lib/gcc/:/opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/lib/../lib/:/opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/../lib/:/opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/../lib/:/opt/conda/envs/gcc/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/lib/:/opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/:/opt/conda/envs/gcc/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/
COLLECT_GCC_OPTIONS='-E' '-v' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
```

Reply:
I guess `-print-sysroot` will include $BUILD_PREFIX instead of $PREFIX, which you need to point to the host env.

Reply:
> Is there a reason to prefer one over the other?

Only that it's retrieving the value from the executable, rather than observing the value at one point in time and making a static copy. Probably doesn't make a difference.

Reply:
Ok, so I should probably keep the version based on HOST.

Reply:
Yeah, that seems fine to me.

Reply:
I'm fairly sure we don't need to modify `CLING_INCLUDE_PATHS` at all. It's already patched in `clangdev` [here](https://github.com/conda-forge/clangdev-feedstock/blob/5.x/recipe/build_clangdev.sh#L1-L13).

Reply:
> I'm fairly sure we don't need to modify CLING_INCLUDE_PATHS at all. It's already patched in clangdev here.

But that patch clearly fixes #18, and the version of clangdev picked up is the one that uses that patch.